mypyc/genops.py

```diff
@@ -1856,6 +1926,30 @@ def is_module_member_expr(self, expr: MemberExpr) -> bool:
         return isinstance(expr.expr, RefExpr) and expr.expr.kind == MODULE_REF
 
     def visit_member_expr(self, expr: MemberExpr) -> Value:
+        # First check if this is maybe a final attribute. This needs to be done differently
+        # for class and module attributes to correctly determine fully qualified name.
+        final_var = None
+        if isinstance(expr.expr, RefExpr) and isinstance(expr.expr.node, TypeInfo):
+            # a class attribute
+            sym = expr.expr.node.get(expr.name)
+            if sym and isinstance(sym.node, Var) and sym.node.is_final:
+                final_var = sym.node
+                fullname = '{}.{}'.format(sym.node.info.fullname(), final_var.name())
+                native = expr.expr.node.module_name in self.modules
+        elif self.is_module_member_expr(expr):
+            # a module attribute
+            if isinstance(expr.node, Var) and expr.node.is_final:
+                final_var = expr.node
+                fullname = expr.node.fullname()
+                native = fullname.rpartition('.')[0] in self.modules
+        if final_var is not None:
+            if final_var.final_value is not None:
+                return self.load_final_literal_value(final_var.final_value, expr.line)
+            elif native:
+                return self.load_final_static(fullname,
+                                              self.mapper.type_to_rtype(self.types[expr]),
+                                              expr.line, final_var.name())
```

Comment:
I'd suggest pulling this into two new methods, one that returns an optional final var and one that emits the final var, to avoid so much of `visit_member_expr` being dedicated to a special case.